### ISSUE TITLE
Plugin-in model

### ISSUE THREAD

**Issue by gramster** [freedmand/semantra#38]
I think it would be great if the document format support was done through a plug-in system. I'd like to add support for epub, and could do a PR here, but having a plugin API seems like a better approach. The PDF support could be done that way too, as well as the requested MS Word support, and maybe even things like GitHub issue support.

**Comment by freedmand** [comment]
Agreed. Something I'm targeting for [v0.2](https://github.com/freedmand/semantra/milestone/1). Every aspect should be documented and pluggable. Part of the nuance with the plug-in system for content loaders is building in frontend renderers as well (where it makes sense to have a unique one, e.g. CSV being shown as a spreadsheet).
